protocol/x/clob/keeper/untriggered_conditional_orders.go

```diff
@@ -288,78 +291,132 @@ func (k Keeper) MaybeTriggerConditionalOrders(ctx sdk.Context) (triggeredConditi
+			allTriggeredOrderIds = append(allTriggeredOrderIds, triggered...)
 
-			// Calculate the max allowed range.
-			maxAllowedRange := lib.BigRatMulPpm(currentOraclePriceSubticksRat, marketParam.MinPriceChangePpm)
-			maxAllowedRange.Mul(maxAllowedRange, new(big.Rat).SetUint64(types.ConditionalOrderTriggerMultiplier))
-
-			upperBound := new(big.Rat).Add(currentOraclePriceSubticksRat, maxAllowedRange)
-			lowerBound := new(big.Rat).Sub(currentOraclePriceSubticksRat, maxAllowedRange)
-
-			for _, price := range []types.Subticks{minTradePriceSubticks, maxTradePriceSubticks} {
-				// Clamp the min and max trade prices to the upper and lower bounds.
-				clampedTradePrice := lib.BigRatClamp(
-					new(big.Rat).SetUint64(price.ToUint64()),
-					lowerBound,
-					upperBound,
-				)
-				triggeredOrderIds := untriggeredConditionalOrders.PollTriggeredConditionalOrders(clampedTradePrice)
-				triggeredConditionalOrderIds = append(triggeredConditionalOrderIds, triggeredOrderIds...)
-			}
+			triggered = k.TriggerOrdersWithPrice(ctx, untriggered, clampedMaxTradePrice, perpetualId, metrics.MaxTradePrice)
+			allTriggeredOrderIds = append(allTriggeredOrderIds, triggered...)
 		}
 
 		// Set the modified untriggeredConditionalOrders back on the keeper field.
-		k.UntriggeredConditionalOrders[clobPairId] = untriggeredConditionalOrders
+		k.UntriggeredConditionalOrders[clobPairId] = untriggered
 	}
 
+	return allTriggeredOrderIds
+}
+
+// TriggerOrdersWithPrice triggers all untriggered conditional orders using the given price. It returns
+// a list of order ids that were triggered. This function is called in EndBlocker.
+func (k Keeper) TriggerOrdersWithPrice(
+	ctx sdk.Context,
+	untriggered *UntriggeredConditionalOrders,
+	price *big.Rat,
+	perpetualId uint32,
+	priceType string,
+) (triggeredOrderIds []types.OrderId) {
+	triggeredOrderIds = untriggered.PollTriggeredConditionalOrders(price)
+
+	// Emit metrics.
+	priceFloat, _ := price.Float32()
+	labels := []gometrics.Label{
+		metrics.GetLabelForStringValue(metrics.Type, priceType),
+		metrics.GetLabelForIntValue(metrics.PerpetualId, int(perpetualId)),
+	}
+	telemetry.SetGaugeWithLabels([]string{metrics.ClobConditionalOrderTriggerPrice}, priceFloat, labels)
+
 	// State write - move the conditional order placement in state from untriggered to triggered state.
 	// Emit an event for each triggered conditional order.
-	for _, triggeredConditionalOrderId := range triggeredConditionalOrderIds {
+	for _, orderId := range triggeredOrderIds {
 		k.MustTriggerConditionalOrder(
 			ctx,
-			triggeredConditionalOrderId,
+			orderId,
 		)
 		k.GetIndexerEventManager().AddTxnEvent(
 			ctx,
 			indexerevents.SubtypeStatefulOrder,
 			indexerevents.StatefulOrderEventVersion,
 			indexer_manager.GetBytes(
 				indexerevents.NewConditionalOrderTriggeredEvent(
-					triggeredConditionalOrderId,
+					orderId,
 				),
 			),
 		)
+
+		telemetry.IncrCounterWithLabels(
```

Comment:
pls use new metrics library!

Reply:
also define new non-modular metric name.

JW why we moved this out of `MustTriggerConditionalOrder`?

Reply:
so that we can add price type as a label (without passing it down as an argument)